src/Sulu/Bundle/AdminBundle/Resources/js/components/CircleSelection/ModifiableCircle.js

```diff
@@ -0,0 +1,168 @@
+    filled: boolean,
+    label: string | typeof undefined,
+    left: number,
+    onChange?: (value: SelectionData) => void,
+    onDoubleClick?: () => void,
+    radius: number,
+    resizable: boolean,
+    top: number,
+};
+
+@observer
+class ModifiableCircle extends React.Component<Props> {
+    @observable moveMode = false;
+    @observable resizeMode = false;
+    @observable resizeAngle = 0;
+
+    static defaultProps = {
+        disabled: false,
+        filled: false,
+        left: 0,
+        radius: 0,
+        resizable: true,
+        top: 0,
+    };
+
+    circleRef: ?ElementRef<'div'>;
+
+    componentDidMount() {
+        window.addEventListener('mouseup', this.handleMouseUp);
+        window.addEventListener('mousemove', this.handleMouseMove);
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('mouseup', this.handleMouseUp);
+        window.removeEventListener('mousemove', this.handleMouseMove);
+    }
+
+    setCircleRef = (ref: ?ElementRef<'div'>) => {
+        this.circleRef = ref;
+    };
+
+    @action handleMoveMouseDown = (event: MouseEvent) => {
+        event.stopPropagation();
+        this.moveMode = true;
+    };
+
+    @action handleResizeMouseDown = (event: MouseEvent) => {
+        event.stopPropagation();
+        this.resizeMode = true;
+    };
+
+    @action handleMouseUp = () => {
+        this.moveMode = false;
+        this.resizeMode = false;
+    };
+
+    @action handleMouseMove = (event: MouseEvent) => {
+        const {onChange, radius, left, top} = this.props;
+
+        if (this.moveMode) {
+            const {movementX, movementY} = event;
+
+            if (!onChange) {
+                return;
+            }
+
+            onChange({
+                radius,
+                left: left + movementX,
+                top: top + movementY,
+            });
+        }
+
+        if (this.resizeMode) {
+            if (!this.circleRef) {
+                return;
+            }
+
+            const rect = this.circleRef.getBoundingClientRect();
+
+            const circleX = rect.left + rect.width / 2;
+            const circleY = rect.top + rect.height / 2;
+            const {clientX: mouseX, clientY: mouseY} = event;
+
+            const deltaX = mouseX - circleX;
+            const deltaY = mouseY - circleY;
+
+            this.resizeAngle = (Math.atan2(deltaY, deltaX) * 180) / Math.PI;
+
+            if (!onChange) {
+                return;
+            }
+
+            const newRadius = Math.sqrt(deltaX ** 2 + deltaY ** 2);
+
+            onChange({
+                left,
+                top,
+                radius: newRadius,
+            });
+        }
+    };
+
+    handleDoubleClick = this.props.onDoubleClick;
+
+    render() {
+        const {disabled, filled, resizable, label, radius, left, top} = this.props;
+        const width = !resizable && radius === 0 ? 30 : radius * 2;
+        const labelSize = radius === 0 ? 14 : Math.sqrt(radius) * 5;
```

Comment:
As far as I remember, it will be bigger than the circle, if the circle is very small, but otherwise it wouldn't be readable anymore. And always using a percentage value has the disadvantage, that it will be really really big on big circles.